[venus-gateway] 启动时配置 --rate-limit-redis 会panic

### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/venus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [venus forum](https://github.com/filecoin-project/venus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/venus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/venus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/venus/issues) and the [venus forum](https://github.com/filecoin-project/venus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`latest release`](https://github.com/filecoin-project/venus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [ ] I did not make any code changes to venus.

### Venus component

- [ ] venus daemon - chain sync
- [ ] venus auth - authentication
- [X] venus gateway - chain service gateway
- [ ] venus miner - mining and block production
- [ ] venus sealer/worker - sealing
- [ ] venus sealer - proving(WindowPoSt)
- [ ] venus market - storage deal
- [ ] venus market - retrieval deal
- [ ] venus market - data transfer
- [ ] venus light-weight client
- [ ] venus JSON-RPC API
- [ ] venus messager - message management (mpool)
- [ ] Other

### Venus Version

```text
v1.6.0
```


### Describe the Bug

启动时配置 --rate-limit-redis 会panic

### Logging Information

```text
2022-06-30T15:00:54.600+0800    INFO    main    venus-gateway/main.go:115       Setting up control endpoint at /ip4/0.0.0.0/tcp/45132
panic: reflect.Value.Addr of unaddressable value

goroutine 1 [running]:
reflect.Value.Addr({0x0?, 0x0?, 0x203000?})
        /usr/local/go/src/reflect/value.go:273 +0x65
github.com/ipfs-force-community/metrics/ratelimit.getInternalStructs({0x2602d20?, 0xc0001fdd60?, 0x69bd2e?})
        /root/go/pkg/mod/github.com/ipfs-force-community/metrics@v1.0.1-0.20211228055608-9462dc86e157/ratelimit/proxycall.go:16 +0x6a
github.com/ipfs-force-community/metrics/ratelimit.GetInternalStructs({0x266f3e0?, 0xc0001fdd60?})
        /root/go/pkg/mod/github.com/ipfs-force-community/metrics@v1.0.1-0.20211228055608-9462dc86e157/ratelimit/proxycall.go:9 +0xaa
github.com/ipfs-force-community/metrics/ratelimit.(*RateLimiter).ProxyLimitFullAPI(0xc0002b84d0, {0x266f3e0?, 0xc0001fdcc0}, {0x266f3e0?, 0xc0001fdd60?})
        /root/go/pkg/mod/github.com/ipfs-force-community/metrics@v1.0.1-0.20211228055608-9462dc86e157/ratelimit/jsonrpcWarper.go:156 +0x65
main.RunMain({0x29358f0?, 0xc000036030}, 0xc00063fbf8)
        /opt/venus-gateway/main.go:131 +0x699
main.glob..func1(0xc0002a07c0)
        /opt/venus-gateway/main.go:91 +0x1a7
github.com/urfave/cli/v2.(*Command).Run(0x51e8000, 0xc0002a05c0)
        /root/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/command.go:163 +0x5bb
github.com/urfave/cli/v2.(*App).RunContext(0xc000103520, {0x29358f0?, 0xc000036030}, {0xc000032090, 0x9, 0x9})
        /root/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:313 +0xb48
github.com/urfave/cli/v2.(*App).Run(...)
        /root/go/pkg/mod/github.com/urfave/cli/v2@v2.3.0/app.go:224
main.main()
        /opt/venus-gateway/main.go:66 +0x23c
```


### Repo Steps

1. Run '...'
2. Do '...'
3. See error '...'
...

--- a/go.mod
+++ b/go.mod
@@ -12,7 +12,7 @@ require (
 	github.com/gbrlsnchs/jwt/v3 v3.0.1
 	github.com/google/uuid v1.3.0
 	github.com/gorilla/mux v1.8.0
-	github.com/ipfs-force-community/metrics v1.0.1-0.20211228055608-9462dc86e157
+	github.com/ipfs-force-community/metrics v1.0.1-0.20220516020337-2f3bf9a2c9e3
 	github.com/ipfs/go-cid v0.1.0
 	github.com/ipfs/go-log/v2 v2.5.1
 	github.com/modern-go/reflect2 v1.0.2
@@ -98,7 +98,6 @@ require (
 	github.com/jinzhu/now v1.1.2 // indirect
 	github.com/json-iterator/go v1.1.12 // indirect
 	github.com/klauspost/cpuid/v2 v2.0.12 // indirect
-	github.com/kr/text v0.2.0 // indirect
 	github.com/leodido/go-urn v1.2.1 // indirect
 	github.com/libp2p/go-buffer-pool v0.0.2 // indirect
 	github.com/libp2p/go-libp2p-core v0.15.1 // indirect
@@ -143,7 +142,6 @@ require (
 	github.com/whyrusleeping/cbor-gen v0.0.0-20220323183124-98fa8256a799 // indirect
 	github.com/whyrusleeping/go-logging v0.0.1 // indirect
 	go.uber.org/atomic v1.9.0 // indirect
-	go.uber.org/goleak v1.1.12 // indirect
 	go.uber.org/multierr v1.8.0 // indirect
 	golang.org/x/crypto v0.0.0-20220411220226-7b82a4e95df4 // indirect
 	golang.org/x/sync v0.0.0-20210220032951-036812b2e83c // indirect

--- a/integrate_test/ratelimit_test.go
+++ b/integrate_test/ratelimit_test.go
@@ -1,0 +1,25 @@
+package integrate
+
+import (
+	"context"
+	"testing"
+
+	"github.com/filecoin-project/go-address"
+	"github.com/ipfs-force-community/venus-gateway/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRateLimit(t *testing.T) {
+	ctx := context.Background()
+
+	cfg := &types.Config{
+		Listen:         "/ip4/127.0.0.1/tcp/0",
+		AuthUrl:        "127.0.0.1:1",
+		JaegerProxy:    "",
+		TraceSampler:   0,
+		TraceNodeName:  "",
+		RateLimitRedis: "127.0.0.1:6379",
+	}
+	_, _, err := MockMain(ctx, []address.Address{}, cfg, defaultTestConfig())
+	require.NoError(t, err)
+}

